File: server/Models/Vault.cs
using System.ComponentModel.DataAnnotations;

namespace keepr.Models;

public class Vault : RepoItems<int>
{
  public string Name { get; set; }
  [MaxLength(3000)] public string Description { get; set; }
  [Url, MaxLength(1000)] public string Img { get; set; }
  public bool isPrivate { get; set; }
  public Profile Creator { get; set; }
  public CreatorId { get; set; }
}

File: client/src/models/Vault.js
import { Profile } from "./Account.js";
import { DatabaseItem } from "./DatabaseItem.js";

export class Vault extends DatabaseItem {
  constructor(data) {
    super(data)
    this.name = data.name
    this.description = data.description
    this.img = data.img
    this.isPrivate = data.isPrivate
    this.creatorId = data.creatorId
    this.creator = new Profile(data.creator)

  }
}